### PR TITLE
hunter.how - new api provider support 

### PR DESCRIPTION
**Pull request by xm1k3**
```
./uncover -e hunterhow -q 'product.name="jira"' -f host -v  

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /    
\__,_/_/ /_/\___/\____/|___/\___/_/ v1.0.2

                projectdiscovery.io

[hunterhow] community.deiser.com
[hunterhow] jira.deiser.com
[hunterhow] retchat.com
[hunterhow] www.retchat.com
[hunterhow] www.rico-baby.com
[hunterhow] ip74.ip-79-137-122.eu
[hunterhow] team.gitee.cn
```

query guide:
https://hunter.how/guide

### CODE REVIEW

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
- [x] Please update the readme with newly added sources (including publicwww)
- [x] Could you provide an example query, I'm getting no results with a valid token and
```console
$ go run . -q test -e hunterhow

```

**Comment by xm1k3** [comment]
```
./uncover -e hunterhow -q 'product.name="jira"' -f host -v  

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /    
\__,_/_/ /_/\___/\____/|___/\___/_/ v1.0.2

                projectdiscovery.io

[hunterhow] community.deiser.com
[hunterhow] jira.deiser.com
[hunterhow] retchat.com
[hunterhow] www.retchat.com
[hunterhow] www.rico-baby.com
[hunterhow] ip74.ip-79-137-122.eu
[hunterhow] team.gitee.cn
```

**Review by Mzack9999 — APPROVED** [review]
lgtm

```console
$ go run . -e hunterhow -q 'product.name=\"jira\"' -f host -v  
...
[hunterhow] dev.proaimltd.com.cn
[hunterhow] build.bmb1688.com
```

File: runner/options.go [hunk 1]
```diff
@@ -142,7 +144,8 @@ func ParseOptions() *Options {
 		len(options.ZoomEye) == 0 &&
 		len(options.Netlas) == 0 &&
 		len(options.CriminalIP) == 0 &&
-		len(options.Publicwww) == 0 {
+		len(options.Publicwww) == 0 &&
+		len(options.HunterHow) == 0 {
```
Comment (tarunKoyalwar) [inline runner/options.go:148]: @xm1k3 , I think we can use newly added generic comparison helper function here https://github.com/projectdiscovery/utils/issues/90

Ex:
```go
genericutils.EqualsAll(0,len(options.ZoomEye),len(options.Netlas))
```

File: uncover/agent/hunterhow/hunterhow.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+			if len(hunterhowResponse) == 0 {
+				break
+			}
+
+			numberOfResults += len(hunterhowResponse)
+			pageQuery += 1
+		}
+	}()
+
```
Comment (tarunKoyalwar) [inline uncover/agent/hunterhow/hunterhow.go:71]: @xm1k3 ,not sure if you want to implement in this PR or a new one . but just to point out this implementation looks good if no of results < 1000 (10 requests are made in serial) but if there are more than 1000> requests it takes time which shouldn't be the case . 
From api docs  it returns ` "total":5994` in response so after first request we can get total results and calculate number of pages available and fetch them according to `limit` cli option in parallel, this way it won't take that much time 

```
so query returns 5994 results it is ~60 req 
60 req in serial (current implementation) takes ~40-50 sec 
60 req in parallel takes ~10 sec
*estimated time here is approx and depends on various factors
```

**Review by tarunKoyalwar — APPROVED** [review]
lgtm ! suggested some possible optimizations
```console
$ ./uncover -hh "product.name=\"jira\"" -f host -v

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /    
\__,_/_/ /_/\___/\____/|___/\___/_/ v1.0.2

		projectdiscovery.io

[hunterhow] gentlecode.com
[hunterhow] www.gentlecode.com
[hunterhow] host64.ru
[hunterhow] jira.erbanyy.com
[hunterhow] jira.ptponclick.com

```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**